File: using_js/sorting.js
function smallest_fn(arr){
    let min = 0;
    for(let i=1; i<=arr.length-1; i++){
        if (arr[min]>arr[i]){
            min = i
        }
    }
    return min
}

function sorting_fn(arr){
    let len = arr.length
    let sorted = []
    let smallest = smallest_fn(arr)
    while (sorted.length != len){
        sorted.push(arr[smallest])
        arr = arr.filter( item => item != arr[smallest])
        smallest = smallest_fn(arr)
        arr.forEach((element, i) => {
            if(element==arr[smallest] && i!=smallest){
                sorted.push(element)
            }
        })
    }
    return sorted
}

console.log(sorting_fn([-1, -2, -3, 1, 4, 8, 9, 7, 3, 9, 8, 1, 2]))

File: using_rs/sorting.rs
fn smallest_fn(arr: &Vec<i64>) -> usize{
    let mut min = 0;
    for i in 0..arr.len() {
        if arr[i]<arr[min]{
            min = i;
        }
    }
    min
}

fn sorting_fn(arr: &[i64]) -> Vec<i64>{
    let mut sorted = Vec::new();
    let mut arr = arr.to_vec();
    let mut smallest = smallest_fn(&arr);
    while !arr.is_empty() {
        sorted.push(arr[smallest]);
        arr = arr.iter().filter(|&item| item != &arr[smallest]).cloned().collect();
        smallest = smallest_fn(&arr);
        for (index, item) in arr.iter().enumerate() {
            if *item == arr[smallest] && index != smallest {
                sorted.push(*item);
            }
        }
    }
    sorted
}

fn main(){
    let arr = [-1, -2, 1, 4, 8, 7, 3, 1, 2];
    println!("{:?} is sorted", sorting_fn(&arr));
}